Version 5.1.5 - multiple versions of @types/react causes typescript build errors

From version 5.1.5 I get errors building my typescript project due to multiple versions of @types/react. I'm not sure if this is a problem with html-react-parser or with @types/react.

See https://github.com/remarkablemark/html-react-parser/commit/78b7a8e8a52b23676a5d86ef60ee9c57683839e3

## Expected Behavior

Should build correctly, possibly should have dependency on @types/react version ^18.2.55 instead of 18.2.55

## Actual Behavior

```
node_modules/@types/react/index.d.ts:4127:14 - error TS2300: Duplicate identifier 'ElementType'.

4127         type ElementType = string | React.JSXElementConstructor<any>;
                  ~~~~~~~~~~~

  node_modules/html-react-parser/node_modules/@types/react/index.d.ts:4127:14
    4127         type ElementType = string | React.JSXElementConstructor<any>;
                      ~~~~~~~~~~~
    'ElementType' was also declared here.

node_modules/@types/react/index.d.ts:4141:14 - error TS2300: Duplicate identifier 'LibraryManagedAttributes'.

4141         type LibraryManagedAttributes<C, P> = C extends
                  ~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/html-react-parser/node_modules/@types/react/index.d.ts:4141:14
    4141         type LibraryManagedAttributes<C, P> = C extends
                      ~~~~~~~~~~~~~~~~~~~~~~~~
    'LibraryManagedAttributes' was also declared here.

node_modules/html-react-parser/node_modules/@types/react/index.d.ts:4127:14 - error TS2300: Duplicate identifier 'ElementType'.

4127         type ElementType = string | React.JSXElementConstructor<any>;
                  ~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:4127:14
    4127         type ElementType = string | React.JSXElementConstructor<any>;
                      ~~~~~~~~~~~
    'ElementType' was also declared here.

node_modules/html-react-parser/node_modules/@types/react/index.d.ts:4141:14 - error TS2300: Duplicate identifier 'LibraryManagedAttributes'.

4141         type LibraryManagedAttributes<C, P> = C extends
                  ~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:4141:14
    4141         type LibraryManagedAttributes<C, P> = C extends
                      ~~~~~~~~~~~~~~~~~~~~~~~~
    'LibraryManagedAttributes' was also declared here.
```


## Steps to Reproduce

git clone git@github.com:captain-igloo/html-react-parser-bug.git
cd html-react-parser-bug
npm ci
npm run build

## Reproducible Demo

https://github.com/captain-igloo/html-react-parser-bug

## Keywords

@types/react

--- a/package.json
+++ b/package.json
@@ -48,7 +48,7 @@
     "dom"
   ],
   "dependencies": {
-    "@types/react": "18.2.55",
+    "@types/react": "17 || 18",
     "domhandler": "5.0.3",
     "html-dom-parser": "5.0.8",
     "react-property": "2.0.2",

